regionmask/core/io.py

```diff
@@ -0,0 +1,101 @@
+import io
+import os
+
+import yaml
+from cartopy.io import Downloader
+
+_default_cache_dir = os.sep.join(("~", ".regionmask_data"))
+longdir_cache = os.path.expanduser(_default_cache_dir)
+if not os.path.exists(longdir_cache):
+    os.mkdir(longdir_cache)
+
+fn_download_yaml = "regionmask/defined_regions/downloadable_regions.yaml"
+with open(fn_download_yaml, "r") as f:
+    metadata = yaml.safe_load(f)
+    keywords_dict = {}
+    for k, v in metadata.items():
+        keywords_dict[k] = k
+        if v["keywords"] is not None:
+            for keyword in v["keywords"]:
+                keywords_dict.update({keyword: k})
+
+with open(fn_download_yaml) as f:
+    download_regions_config = yaml.safe_load(f)
+f.close()
+
+
+class RegionmaskDownloader(Downloader):
+    """Copied from cartopy.io.shapereader. Close to NEShpDownloader."""
+
+    def zip_file_contents(self, format_dict):
+        """
+        Return a generator of the filenames to be found in the downloaded
+        natural earth zip file.
+        """
+        for ext in [".shp", ".dbf", ".shx"]:
+            # this line got changed from cartopy
```

Comment:
because of this one line I couldnt take `NEShpDownloader` from `cartopy` but I copied all other lines